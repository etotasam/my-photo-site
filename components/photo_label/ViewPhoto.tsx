import React, { useState, useEffect } from "react";
import NextImage from "next/image";
import { useRouter } from "next/router";
import { AnimatePresence, motion } from "framer-motion";
import Loading from "../Loading";
import { ImagesType } from "@/assets/type/types";
import { useHeadersContext, InitialState } from "@/components/header/HeadersContext";

type State = {
  state: InitialState;
  dispatch: React.Dispatch<any>;
}

type Params = {
  imageRef: ImagesType;
  length: number;
  element: React.MutableRefObject<any>
};


const ViewPhoto = ({ imageRef, length, element }: Params) => {
  const router = useRouter();
  const { photo_label, image } = router.query;
  function prevPhoto() {
    let prev: number;
    if (image) prev = Number(image) - 1;
    if (!image) prev = length;
    if (prev < 1) return router.push(`/photo/${photo_label}?image=${length}`);
    router.push(`/photo/${photo_label}?image=${prev}`);
  }

  // get values headerHeight and footerHeight by useContext
  const { state: contextState, dispatch }: State = useHeadersContext();

  // check window aspect ratio
  const [imageStyle, setImageStyle] = useState<string>()
  const adjustContainerToImage = (): void => {
    const headerHeight = contextState.headerHeight;
    const footerHeight = contextState.footerHeight;
    const windowHeight = window.innerHeight;
    const containerWidth: number = element.current.clientWidth;
    // const containerHeight = element.current.clientHeight;
    const containerHeight: number = windowHeight - (headerHeight + footerHeight);

    const isWindowHorizontal: boolean = containerWidth > containerHeight;

    const imgWidth = imageRef.width;
    const imgHeight = imageRef.height;

    const isImageHorizontal = imgWidth > imgHeight;

    let maxImgWidth: number;
    let maxImgHeight: number;
    const maxImageLongSideNum = 850;
    const minImageLongSideNum = 350;
    document.documentElement.style.setProperty(`--min-img-long-side`, `${minImageLongSideNum}px`)
    if(isImageHorizontal) {
      const aspectRatio = imgHeight / imgWidth;
      const maxHeight = maxImageLongSideNum * aspectRatio
      maxImgWidth = Math.min(imgWidth, maxImageLongSideNum)
      maxImgHeight = Math.min(imgHeight, maxHeight)
    }else {
      const aspectRatio = imgWidth / imgHeight
      const maxWidth = maxImageLongSideNum * aspectRatio;
      maxImgWidth = Math.min(imgWidth, maxWidth)
      maxImgHeight = Math.min(imgHeight, maxImageLongSideNum)
    }


    document.documentElement.style.setProperty(`--img-container-width`, `${containerWidth}px`)
    // document.documentElement.style.setProperty(`--img-container-height`, `${containerHeight}px`)
    document.documentElement.style.setProperty(`--img-width`, `${imgWidth}px`)
    document.documentElement.style.setProperty(`--img-width`, `${imgHeight}px`)

    document.documentElement.style.setProperty(`--max-img-width`, `${maxImgWidth}px`)
    document.documentElement.style.setProperty(`--max-img-height`, `${maxImgHeight}px`)

    const aspectRatioForHeight = imgHeight / imgWidth;
    const aspectRatioForWidth = imgWidth / imgHeight;
    document.documentElement.style.setProperty(`--img-aspect-ratio-for-height`, `${aspectRatioForHeight}`)
    document.documentElement.style.setProperty(`--img-aspect-ratio-for-width`, `${aspectRatioForWidth}`)

    const isContainerWidthLarge = containerWidth > maxImgWidth
    const isContainerHeightLarge = containerHeight > maxImgHeight

    const setCssClass = () => {
      if(isImageHorizontal) {
        if(isContainerWidthLarge && isContainerHeightLarge) return `t-img_conrainer_horizon-both-high`;
        if(isContainerWidthLarge && !isContainerHeightLarge) return `t-img_container_horizon-width_high-height_low`;
        if(!isContainerWidthLarge && isContainerHeightLarge) return `t-img_container_horizon-width_low-height_high`;
        if(!isContainerWidthLarge && !isContainerHeightLarge) return isWindowHorizontal ? `t-img_container_horizon-both_low-window_horizon` : `t-img_container_horizon-both_low-window_vertical`;
      }else {
        if(isContainerWidthLarge && isContainerHeightLarge) return `t-img_container_vertical-both_high`;
        if(isContainerWidthLarge && !isContainerHeightLarge) return `t-img_container_vertical-width_high-height_low`;
        if(!isContainerWidthLarge && isWindowHorizontal) return `t-img_container_vertical-width_low-window_horizon`;
        if(!isContainerWidthLarge && !isWindowHorizontal) return `t-img_container_vertical-width_low-window_vertical`;
      }
    }
    setImageStyle(setCssClass())
  }
  useEffect(() => {
    adjustContainerToImage()
    window.addEventListener(`resize`, adjustContainerToImage)
    return () => {
      window.removeEventListener(`resize`, adjustContainerToImage)
    }
  }, [])

  // photo transition
  function nextPhoto() {
    let next: number;
    if (image) next = Number(image) + 1;
    if (!image) next = 2;
    if (next > length) return router.push(`/photo/${photo_label}?image=1`);
    router.push(`/photo/${photo_label}?image=${next}`);
  }

  // check image aspect ratio
  const [isVerticalPhoto, setIsPhotoVertical] = useState<boolean>();
  useEffect(() => {
    if (!imageRef) return;
    setIsPhotoVertical(imageRef.width < imageRef.height);
  }, [imageRef]);

  // tap reaction
  let tapPositionX: number;
  let unTapPositionX: number;
  function onTapStart(event: any, info: any) {
    tapPositionX = info.point.x;
  }

  const necessaryMoveX = 100;
  function onTap(event: any, info: any) {
    unTapPositionX = info.point.x;
    const movedPositionX = unTapPositionX - tapPositionX;

    if (movedPositionX < -necessaryMoveX) {
      nextPhoto();
      return;
    } else if (necessaryMoveX < movedPositionX) {
      prevPhoto();
      return;
    }
  }

  // image loading check
  const [isImageLoading, setImageLoad] = useState(true);
  function photoLoaded() {
    setImageLoad(false);
  }

  // image transtion
  const imageTranstion = (e: any) => {
    const elWidth = e.target.clientWidth;
    const elPosition = Math.floor(e.target.getBoundingClientRect().left);
    const centerPoint = elWidth / 2;
    const clickPoint = e.pageX;
    if((clickPoint - elPosition) > centerPoint) {
      nextPhoto();
    }else {
      prevPhoto();
    }
  }

  return (
    <>
      <motion.div
        initial={{ opacity: 0 }}
        animate={{ opacity: 1 }}
        onTapStart={onTapStart}
        onTap={onTap}
        transition={{ duration: 0.5 }}
        className={`relative leading-3 bg-green-200 ${imageStyle}`}
        style={{ touchAction: "none" }}
        onClick={(e) => imageTranstion(e)}
      >
        <NextImage
          // onClick={(e) => click(e)}
          className={`cursor-pointer`}
          src={imageRef.url}
          width={imageRef.width}
          height={imageRef.height}
          alt={``}
          priority={true}
          layout={`fill`}
          objectFit={`contain`}
          onLoad={photoLoaded}
        />
        <AnimatePresence>{isImageLoading && <Loading />}</AnimatePresence>
      </motion.div>
    </>
  );
};

export default ViewPhoto;
