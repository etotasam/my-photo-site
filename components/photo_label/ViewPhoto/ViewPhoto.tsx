import React, { useState, useEffect } from "react";
import NextImage from "next/image";
import { useRouter } from "next/router";
import { AnimatePresence, motion } from "framer-motion";
import type { ImagesType } from "@/@types/types";
import { useModalDispatchContext } from "@/context/modalStateContext";
import { useHeihgtStateContext } from "@/context/heightStateContext";
import { useLoadDispatchContext } from "@/context/loadStateContext";
import { useViewPhotoSize } from "@/hooks";
//! component
import { Loading } from "@/components/Loading";
import { LoadingBound } from "@/components/LoadingBound";

type Params = {
  imageRef: ImagesType;
  imagesLength: number;
};

export const ViewPhoto = ({ imageRef, imagesLength: lastImage }: Params) => {
  const router = useRouter();
  const { photo_label, image } = router.query;
  const { headerHeight, footerHeight } = useHeihgtStateContext();
  const size = useViewPhotoSize(imageRef, headerHeight, footerHeight);

  function prevPhoto() {
    let prev: number | undefined;
    if (image) prev = Number(image) - 1;
    if (!image) prev = lastImage;
    if (prev! < 1) return router.push(`/photo/${photo_label}?image=${lastImage}`);
    router.push(`/photo/${photo_label}?image=${prev}`);
  }

  // photo transition
  function nextPhoto() {
    let next: number;
    if (image) next = Number(image) + 1;
    if (!image) next = 2;
    if (next! > lastImage) return router.push(`/photo/${photo_label}?image=1`);
    router.push(`/photo/${photo_label}?image=${next!}`);
  }

  // image loading check
  const [isImageLoading, setIsImageLoading] = useState<boolean>(true);
  function closeLoadingModal() {
    setIsImageLoading(false);
  }

  // get values headerHeight and footerHeight by useContext
  const { modalCloseDispatcher } = useModalDispatchContext();
  const { loadedDispatcher } = useLoadDispatchContext();

  const closeModal = () => {
    modalCloseDispatcher();
    loadedDispatcher();
  };

  useEffect(() => {
    // imageLoad();
    closeModal();
  }, []);

  // image transtion
  const imageTranstion = (e: React.MouseEvent<HTMLDivElement, MouseEvent>) => {
    const el = e.target as HTMLElement;
    const elWidth = el.clientWidth;
    const elLeftPosition = Math.floor(el.getBoundingClientRect().left);
    const centerPoint = elWidth / 2;
    const clickPoint = e.pageX;
    if (clickPoint - elLeftPosition > centerPoint) {
      nextPhoto();
    } else {
      prevPhoto();
    }
  };

  let tapOnPositon: number;
  const tapOn = (e: React.TouchEvent<HTMLImageElement>) => {
    tapOnPositon = e.changedTouches[0].pageX;
  };

  const requireNum = 100;
  const tapOff = (e: React.TouchEvent<HTMLImageElement>) => {
    const tapOffPosition = e.changedTouches[0].pageX;
    const movedNum = tapOffPosition - tapOnPositon;
    if (Math.abs(movedNum) < requireNum) return;
    if (movedNum > 0) {
      nextPhoto();
    } else {
      prevPhoto();
    }
  };

  return (
    <>
      <motion.div
        data-testid={`imageWrap`}
        initial={{ opacity: 0 }}
        animate={{ opacity: 1 }}
        onTouchStart={tapOn}
        onTouchEnd={tapOff}
        transition={{ duration: 0.5 }}
        className={`relative leading-3`}
        style={{ ...size }}
        onClick={(e) => imageTranstion(e)}
      >
        <NextImage
          className={`cursor-pointer`}
          src={imageRef.url}
          alt={``}
          priority={true}
          layout={`fill`}
          objectFit={`contain`}
          onLoad={closeLoadingModal}
        />
      </motion.div>
      <AnimatePresence>{isImageLoading && <Loading />}</AnimatePresence>
    </>
  );
};
