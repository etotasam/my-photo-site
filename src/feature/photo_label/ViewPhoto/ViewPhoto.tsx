import React from "react";
import NextImage from "next/image";

import { AnimatePresence, motion } from "framer-motion";
import type { ImagesType } from "@/@types/types";
//! context

import { useHeihgtStateContext } from "@/context/heightStateContext";

//! hooks
import { useAdjustSizeForWrapperPhoto } from "@/hooks";
//! component
import { Loading } from "@/feature/Loading";

type Params = {
  imageRef: ImagesType;
  imageTranstion: (e: React.MouseEvent<HTMLDivElement, MouseEvent>) => void;
  tapOn: (e: React.TouchEvent<HTMLImageElement>) => void;
  tapOff: (e: React.TouchEvent<HTMLImageElement>) => void;
  isImageLoading: boolean;
  closeLoadingModal: () => void;
};

export const ViewPhoto = ({ imageRef, imageTranstion, tapOn, tapOff, isImageLoading, closeLoadingModal }: Params) => {
  const { headerHeight, footerHeight } = useHeihgtStateContext();
  const { photoSize } = useAdjustSizeForWrapperPhoto(imageRef, headerHeight, footerHeight);

  return (
    <>
      <motion.div
        data-testid={`imageWrap`}
        initial={{ opacity: 0 }}
        animate={{ opacity: 1 }}
        onTouchStart={tapOn}
        onTouchEnd={tapOff}
        transition={{ duration: 0.5 }}
        className={`relative leading-3`}
        style={{ ...photoSize }}
        onClick={(e) => imageTranstion(e)}
      >
        <NextImage
          className={`cursor-pointer`}
          src={imageRef.url}
          alt={``}
          priority={true}
          layout={`fill`}
          objectFit={`contain`}
          onLoad={closeLoadingModal}
        />
      </motion.div>
      <AnimatePresence>{isImageLoading && <Loading />}</AnimatePresence>
    </>
  );
};
