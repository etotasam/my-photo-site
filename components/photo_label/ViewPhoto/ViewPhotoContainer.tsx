import React, { useState, useEffect } from "react";
import { useRouter } from "next/router";
//! component
import { ViewPhoto } from "./ViewPhoto";

//! types
import type { ImagesType } from "@/@types/types";

//! context
import { useModalDispatchContext } from "@/context/modalStateContext";
import { useHeihgtStateContext } from "@/context/heightStateContext";
import { useLoadDispatchContext } from "@/context/loadStateContext";

type Params = {
  imageRef: ImagesType;
  imagesLength: number;
};
export const ViewPhotoContainer = ({ imageRef, imagesLength: lastImage }: Params) => {
  const router = useRouter();
  const { photo_label, image } = router.query;

  //? photo transition
  const prevPhoto = () => {
    let prev: number | undefined;
    if (image) prev = Number(image) - 1;
    if (!image) prev = lastImage;
    if (prev! < 1) return router.push(`/photo/${photo_label}?image=${lastImage}`);
    router.push(`/photo/${photo_label}?image=${prev}`);
  };

  //? photo transition
  const nextPhoto = () => {
    let next: number;
    if (image) next = Number(image) + 1;
    if (!image) next = 2;
    if (next! > lastImage) return router.push(`/photo/${photo_label}?image=1`);
    router.push(`/photo/${photo_label}?image=${next!}`);
  };

  //? get values headerHeight and footerHeight by useContext
  const { modalCloseDispatcher } = useModalDispatchContext();
  const { loadedDispatcher } = useLoadDispatchContext();

  const closeModal = () => {
    modalCloseDispatcher();
    loadedDispatcher();
  };

  useEffect(() => {
    //? imageLoad();
    closeModal();
  }, []);

  //? image transtion
  const imageTranstion = (e: React.MouseEvent<HTMLDivElement, MouseEvent>) => {
    const el = e.target as HTMLElement;
    const elWidth = el.clientWidth;
    const elLeftPosition = Math.floor(el.getBoundingClientRect().left);
    const centerPoint = elWidth / 2;
    const clickPoint = e.pageX;
    if (clickPoint - elLeftPosition > centerPoint) {
      nextPhoto();
    } else {
      prevPhoto();
    }
  };

  const tapOnPositon = React.useRef<number>();
  const tapOn = (e: React.TouchEvent<HTMLImageElement>) => {
    tapOnPositon.current = e.changedTouches[0].pageX;
  };

  const requireNum = 100;
  const tapOff = (e: React.TouchEvent<HTMLImageElement>) => {
    if (!tapOnPositon.current) {
      if (process.env.NEXT_PUBLIC_IS_DEV) {
        console.error("tapOnPositon is null");
      }
      return;
    }
    const tapOffPosition = e.changedTouches[0].pageX;
    const movedNum = tapOffPosition - tapOnPositon.current;
    if (Math.abs(movedNum) < requireNum) return;
    if (movedNum > 0) {
      nextPhoto();
    } else {
      prevPhoto();
    }
  };

  //? image loading check
  const [isImageLoading, setIsImageLoading] = useState<boolean>(true);
  const closeLoadingModal = () => {
    setIsImageLoading(false);
  };
  return (
    <ViewPhoto
      imageRef={imageRef}
      imageTranstion={imageTranstion}
      tapOn={tapOn}
      tapOff={tapOff}
      isImageLoading={isImageLoading}
      closeLoadingModal={closeLoadingModal}
    />
  );
};
