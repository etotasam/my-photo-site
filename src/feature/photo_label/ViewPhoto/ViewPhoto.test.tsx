import React from "react";
import { ViewPhoto } from "./ViewPhoto";
import { fireEvent, render, screen, act } from "@testing-library/react";
import { cleanup } from "@testing-library/react-hooks";
import renderer from "react-test-renderer";

type ImagesType = {
  documentId: string;
  width: number;
  createAt: Date;
  height: number;
  url: string;
  filename: string;
  id: string;
};

const image = {
  documentId: "",
  width: 0,
  createAt: new Date(),
  height: 0,
  url: "https://picsum.photos/seed/picsum/200/300",
  filename: "",
  id: "1",
};

const imageTranstionMock = jest.fn().mockReturnValue("get imageTransiton");

const props = {
  imageRef: image,
  imageTranstion: imageTranstionMock,
  tapOn: jest.fn().mockReturnValue("gat tapOn"),
  tapOff: jest.fn().mockReturnValue("gat tapOff"),
  isImageLoading: false,
  closeLoadingModal: jest.fn().mockReturnValue(true),
};

const nextRouer = jest.spyOn(require("next/router"), `useRouter`);
const query = { photo_label: "jordan", image: "1" };
nextRouer.mockImplementation(() => ({
  query,
}));

afterEach(() => cleanup());

describe(`ViewPhoto`, () => {
  it(`縦長イメージの表示サイズの検証`, async () => {
    image.width = 1200;
    image.height = 800;
    const { asFragment } = render(<ViewPhoto {...props} />);
    act(() => resizeWidth(1800, 1200));
    expect(asFragment()).toMatchSnapshot();
    act(() => resizeWidth(700, 1000));
    expect(asFragment()).toMatchSnapshot();
  });
  it(`横長イメージの表示サイズの検証`, () => {
    image.width = 800;
    image.height = 1200;
    const { asFragment } = render(<ViewPhoto {...props} />);
    act(() => resizeWidth(1800, 1200));
    expect(asFragment()).toMatchSnapshot();
    act(() => resizeWidth(700, 1000));
    expect(asFragment()).toMatchSnapshot();
  });
  it(`画像クリックでonClick指定の関数が呼ばれる`, () => {
    render(<ViewPhoto {...props} />);
    const imageEl = screen.getByTestId(`imageWrap`);
    fireEvent.click(imageEl);
    expect(imageTranstionMock).toBeCalledTimes(1);
  });
});

const resizeWidth = (width: number, height: number) => {
  window.innerWidth = width;
  window.innerHeight = height;
  window.dispatchEvent(new Event("resize"));
};
