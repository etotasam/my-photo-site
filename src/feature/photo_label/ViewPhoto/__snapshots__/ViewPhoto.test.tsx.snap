// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ViewPhoto 横長イメージの表示サイズの検証 1`] = `
<DocumentFragment>
  <div
    class="relative leading-3"
    data-testid="imageWrap"
    style="width: 798px; height: 1198px; min-width: 233px; min-height: 350px; max-width: 666px; max-height: 1000px; opacity: 0;"
  >
    <div
      style="display: block; overflow: hidden; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; margin: 0px;"
    >
      <img
        alt=""
        class="cursor-pointer"
        data-nimg="fill"
        decoding="async"
        sizes="100vw"
        src="/_next/image?url=https%3A%2F%2Fpicsum.photos%2Fseed%2Fpicsum%2F200%2F300&w=3840&q=75"
        srcset="/_next/image?url=https%3A%2F%2Fpicsum.photos%2Fseed%2Fpicsum%2F200%2F300&w=640&q=75 640w, /_next/image?url=https%3A%2F%2Fpicsum.photos%2Fseed%2Fpicsum%2F200%2F300&w=750&q=75 750w, /_next/image?url=https%3A%2F%2Fpicsum.photos%2Fseed%2Fpicsum%2F200%2F300&w=828&q=75 828w, /_next/image?url=https%3A%2F%2Fpicsum.photos%2Fseed%2Fpicsum%2F200%2F300&w=1080&q=75 1080w, /_next/image?url=https%3A%2F%2Fpicsum.photos%2Fseed%2Fpicsum%2F200%2F300&w=1200&q=75 1200w, /_next/image?url=https%3A%2F%2Fpicsum.photos%2Fseed%2Fpicsum%2F200%2F300&w=1920&q=75 1920w, /_next/image?url=https%3A%2F%2Fpicsum.photos%2Fseed%2Fpicsum%2F200%2F300&w=2048&q=75 2048w, /_next/image?url=https%3A%2F%2Fpicsum.photos%2Fseed%2Fpicsum%2F200%2F300&w=3840&q=75 3840w"
        style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: contain;"
      />
      <noscript />
    </div>
  </div>
</DocumentFragment>
`;

exports[`ViewPhoto 横長イメージの表示サイズの検証 2`] = `
<DocumentFragment>
  <div
    class="relative leading-3"
    data-testid="imageWrap"
    style="width: 420px; height: 630px; min-width: 233px; min-height: 350px; max-width: 666px; max-height: 1000px; opacity: 0;"
  >
    <div
      style="display: block; overflow: hidden; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; margin: 0px;"
    >
      <img
        alt=""
        class="cursor-pointer"
        data-nimg="fill"
        decoding="async"
        sizes="100vw"
        src="/_next/image?url=https%3A%2F%2Fpicsum.photos%2Fseed%2Fpicsum%2F200%2F300&w=3840&q=75"
        srcset="/_next/image?url=https%3A%2F%2Fpicsum.photos%2Fseed%2Fpicsum%2F200%2F300&w=640&q=75 640w, /_next/image?url=https%3A%2F%2Fpicsum.photos%2Fseed%2Fpicsum%2F200%2F300&w=750&q=75 750w, /_next/image?url=https%3A%2F%2Fpicsum.photos%2Fseed%2Fpicsum%2F200%2F300&w=828&q=75 828w, /_next/image?url=https%3A%2F%2Fpicsum.photos%2Fseed%2Fpicsum%2F200%2F300&w=1080&q=75 1080w, /_next/image?url=https%3A%2F%2Fpicsum.photos%2Fseed%2Fpicsum%2F200%2F300&w=1200&q=75 1200w, /_next/image?url=https%3A%2F%2Fpicsum.photos%2Fseed%2Fpicsum%2F200%2F300&w=1920&q=75 1920w, /_next/image?url=https%3A%2F%2Fpicsum.photos%2Fseed%2Fpicsum%2F200%2F300&w=2048&q=75 2048w, /_next/image?url=https%3A%2F%2Fpicsum.photos%2Fseed%2Fpicsum%2F200%2F300&w=3840&q=75 3840w"
        style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: contain;"
      />
      <noscript />
    </div>
  </div>
</DocumentFragment>
`;

exports[`ViewPhoto 縦長イメージの表示サイズの検証 1`] = `
<DocumentFragment>
  <div
    class="relative leading-3"
    data-testid="imageWrap"
    style="width: 1198px; height: 798px; min-width: 350px; min-height: 233px; max-width: 1000px; max-height: 666px; opacity: 0;"
  >
    <div
      style="display: block; overflow: hidden; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; margin: 0px;"
    >
      <img
        alt=""
        class="cursor-pointer"
        data-nimg="fill"
        decoding="async"
        sizes="100vw"
        src="/_next/image?url=https%3A%2F%2Fpicsum.photos%2Fseed%2Fpicsum%2F200%2F300&w=3840&q=75"
        srcset="/_next/image?url=https%3A%2F%2Fpicsum.photos%2Fseed%2Fpicsum%2F200%2F300&w=640&q=75 640w, /_next/image?url=https%3A%2F%2Fpicsum.photos%2Fseed%2Fpicsum%2F200%2F300&w=750&q=75 750w, /_next/image?url=https%3A%2F%2Fpicsum.photos%2Fseed%2Fpicsum%2F200%2F300&w=828&q=75 828w, /_next/image?url=https%3A%2F%2Fpicsum.photos%2Fseed%2Fpicsum%2F200%2F300&w=1080&q=75 1080w, /_next/image?url=https%3A%2F%2Fpicsum.photos%2Fseed%2Fpicsum%2F200%2F300&w=1200&q=75 1200w, /_next/image?url=https%3A%2F%2Fpicsum.photos%2Fseed%2Fpicsum%2F200%2F300&w=1920&q=75 1920w, /_next/image?url=https%3A%2F%2Fpicsum.photos%2Fseed%2Fpicsum%2F200%2F300&w=2048&q=75 2048w, /_next/image?url=https%3A%2F%2Fpicsum.photos%2Fseed%2Fpicsum%2F200%2F300&w=3840&q=75 3840w"
        style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: contain;"
      />
      <noscript />
    </div>
  </div>
</DocumentFragment>
`;

exports[`ViewPhoto 縦長イメージの表示サイズの検証 2`] = `
<DocumentFragment>
  <div
    class="relative leading-3"
    data-testid="imageWrap"
    style="width: 630px; height: 420px; min-width: 350px; min-height: 233px; max-width: 1000px; max-height: 666px; opacity: 0;"
  >
    <div
      style="display: block; overflow: hidden; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; margin: 0px;"
    >
      <img
        alt=""
        class="cursor-pointer"
        data-nimg="fill"
        decoding="async"
        sizes="100vw"
        src="/_next/image?url=https%3A%2F%2Fpicsum.photos%2Fseed%2Fpicsum%2F200%2F300&w=3840&q=75"
        srcset="/_next/image?url=https%3A%2F%2Fpicsum.photos%2Fseed%2Fpicsum%2F200%2F300&w=640&q=75 640w, /_next/image?url=https%3A%2F%2Fpicsum.photos%2Fseed%2Fpicsum%2F200%2F300&w=750&q=75 750w, /_next/image?url=https%3A%2F%2Fpicsum.photos%2Fseed%2Fpicsum%2F200%2F300&w=828&q=75 828w, /_next/image?url=https%3A%2F%2Fpicsum.photos%2Fseed%2Fpicsum%2F200%2F300&w=1080&q=75 1080w, /_next/image?url=https%3A%2F%2Fpicsum.photos%2Fseed%2Fpicsum%2F200%2F300&w=1200&q=75 1200w, /_next/image?url=https%3A%2F%2Fpicsum.photos%2Fseed%2Fpicsum%2F200%2F300&w=1920&q=75 1920w, /_next/image?url=https%3A%2F%2Fpicsum.photos%2Fseed%2Fpicsum%2F200%2F300&w=2048&q=75 2048w, /_next/image?url=https%3A%2F%2Fpicsum.photos%2Fseed%2Fpicsum%2F200%2F300&w=3840&q=75 3840w"
        style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: contain;"
      />
      <noscript />
    </div>
  </div>
</DocumentFragment>
`;
